### PR TITLE
feat(mechanics): Add a new personality to make ships with full cargo flee

### LINKED ISSUE #10597: Dumping cargo as escape mechanic
### Problem Description

Merchant cargo ships often dump their cargo and escape the system. It's hard to tell whether or not this affects the behaviour of pirate ships attacking them, but we can assume an in-lore explanation where they ignore the ship in favour of the dumped cargo. Otherwise dumping the cargo would be a dumb, pointless move.

The player has the ability to dump cargo, but as of now, the pirates just go "Hah boys! Looks like he shat himself!" and continue attacking you like nothing happened.

This is strange because the game has an explicit mechanic where your cargo raises the piracy threat. For pirates to just ignore that contradicts the lore.

### Related Issue Links

#5317

### Desired Solution

When you use the 'Dump cargo' button, pirates should stop attacking for a few seconds to collect it.

Ideas:
- There can be an element of randomness to this: only some of the attacking pirates switch to cargo hunting, or maybe there's a single RNG roll upon dumping that decides whether all the pirates attacking you will pause or not.
- The more cargo you have, the bigger the escape window could be (pirates take longer to collect it).
- The more valuable your cargo, the higher the chance that pirates switch their attentions.
- When you selectively jettison cargo, that works too.
- If the number of cargo items are less than the number of attacking pirates, only an equal number of attackers are distracted.

### Alternative Approaches

If you have cargo in your ship, the game could automatically assign a threshold where if your ship is moments away from being disabled, it automatically dumps its cargo in order to create an escape window.

Maybe this happens with mission cargo too. You fail the mission, but at least you survive.

The cargo could be dumped over time instead of all at once: this might work better in some scenarios in terms of creating an escape window by forcing pirates to slow down repeatedly to collect it, instead of in one swooping arc.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Anarchist2**
## Summary
This PR adds a new personality, `getaway`, that makes ships with it jump away if their cargo holds are full. The main reason behind adding this is to make dumping cargo better: right now, ships will only fly to flotsam if their cargo holds aren't full, meaning that pirate ships with full holds won't be distracted by jettisoned cargo.

This PR also adds `getaway` to all existing pirate fleets.

Closes #10597 

## Usage examples
```
fleet "Small Northern Pirates"
	government "Pirate"
	names "pirate"
	cargo 1
	personality
		disables plunders harvests getaway
		confusion 20
```

## Testing Done
Dumped some cargo. Pirates picked it up. Some of the pirates began to run away. Scanned the pirates and saw the fleeing ships had full cargo holds, and the staying pirates either still had cargo space or the lead ship of their ship still had free cargo space.

## Wiki Update
https://github.com/endless-sky/endless-sky-wiki/pull/70

## Performance Impact
`ShouldFlee()` now checks some more stuff.

### CODE REVIEW

File: data/human/fleets.txt
Comment (TheGiraffe3) [inline data/human/fleets.txt:1]: Would it be possible to add this to make the personalities alphabetical?

Comment (TomGoodIdea) [reply data/human/fleets.txt:1]: they weren't alphabetical before

**Review by TomGoodIdea — APPROVED** [review]
code looks ok

**Review by TheGiraffe3 — APPROVED** [review]

**Review by Quantumshark — APPROVED** [review]
Makes sense.

**Pull request merged**